app/src/main/java/com/jerboa/ui/components/blocks/BlocksActivity.kt

```diff
@@ -0,0 +1,275 @@
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.toMutableStateList
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.jerboa.R
+import com.jerboa.api.API
+import com.jerboa.api.ApiState
+import com.jerboa.api.toApiState
+import com.jerboa.model.SiteViewModel
+import com.jerboa.showBlockCommunityToast
+import com.jerboa.showBlockInstanceToast
+import com.jerboa.showBlockPersonToast
+import com.jerboa.ui.components.common.ApiEmptyText
+import com.jerboa.ui.components.common.ApiErrorText
+import com.jerboa.ui.components.common.CircularIcon
+import com.jerboa.ui.components.common.JerboaSnackbarHost
+import com.jerboa.ui.components.common.LoadingBar
+import com.jerboa.ui.components.common.SimpleTopAppBar
+import com.jerboa.ui.components.common.simpleVerticalScrollbar
+import com.jerboa.ui.theme.Title
+import it.vercruysse.lemmyapi.v0x19.datatypes.BlockCommunity
+import it.vercruysse.lemmyapi.v0x19.datatypes.BlockCommunityResponse
+import it.vercruysse.lemmyapi.v0x19.datatypes.BlockInstance
+import it.vercruysse.lemmyapi.v0x19.datatypes.BlockInstanceResponse
+import it.vercruysse.lemmyapi.v0x19.datatypes.BlockPerson
+import it.vercruysse.lemmyapi.v0x19.datatypes.BlockPersonResponse
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+private const val UNBLOCK_BUTTON_SIZE = 18
+
+@Composable
+private fun BlockedElementListItem(
+    apiState: ApiState<*>,
+    icon: String?,
+    name: String,
+    onUnblock: () -> Unit,
+    onSuccessfulUnblock: () -> Unit,
+) {
+    Row(verticalAlignment = Alignment.CenterVertically) {
+        icon?.let {
+            CircularIcon(
+                icon = it,
+                contentDescription = "",
+                size = 26.dp,
+            )
+            Spacer(modifier = Modifier.padding(horizontal = 4.dp))
+        }
+        Text(name, modifier = Modifier.weight(1f))
+        TextButton(onClick = onUnblock, colors = ButtonDefaults.buttonColors(Color.Transparent)) {
+            when (apiState) {
+                ApiState.Loading -> CircularProgressIndicator(
+                    modifier = Modifier.size(UNBLOCK_BUTTON_SIZE.dp),
+                    color = Color.Gray,
+                )
+                is ApiState.Success -> onSuccessfulUnblock()
+                else -> Icon(
+                    imageVector = Icons.Rounded.Close,
+                    modifier = Modifier.size(UNBLOCK_BUTTON_SIZE.dp),
+                    tint = Color.Red,
```

Comment:
I used ``Material.colorScheme.error`` as you suggested, but the "X" icon is blue instead of red, are my app settings wrong?

![image](https://github.com/LemmyNet/jerboa/assets/56604262/85c5c8cf-bb99-4ab2-a103-66dcbbe8f7f2)

Reply:
No it depends on your selected theme. Light + blue theme will have that effect l